### PR TITLE
[DEVOPS-227] enable cardano profiling

### PR DESCRIPTION
**Pull request by cleverca22**
@domenkozar there is now a profiling option, and hydra will pre-build everything with and without profiling

### CODE REVIEW

**Comment by deepfire** [comment]
@cleverca22, looks very nice!

File: default.nix [hunk 1]
```diff
@@ -5,7 +5,9 @@ in
 , config ? {}
 , dconfig ? "testnet_staging_full"
 , gitrev ? "unknown"
-, pkgs ? (import (localLib.fetchNixPkgs) { inherit system config; }) }:
+, pkgs ? (import (localLib.fetchNixPkgs) { inherit system config; })
+, enableProfiling ? false
```
Comment (domenkozar) [inline default.nix]: I'd add a comment here: `# profiling slows down performance by 50% so we don't enable it by default`

File: node/cardano-sl.cabal [hunk 2]
```diff
@@ -25,6 +25,7 @@ Flag with-explorer
   description: Build with explorer
 
 library
+  extra-libraries:      m
```
Comment (domenkozar) [inline node/cardano-sl.cabal]: Our used nixpkgs commit a905b7cd0c2dc0714195a50bf176cd8e4593502d includes the GHC patch in aafe3d29c1572b89aa0f4f8095ffd9a43c77229e - so I wonder why we need this?

Comment (domenkozar) [reply node/cardano-sl.cabal]: Also once we figure out why we need this, it will have to be enabled only for linux

File: release.nix [hunk 3]
```diff
@@ -27,6 +28,7 @@ let
     cardano-sl-explorer-static = [ "x86_64-linux" ];
     cardano-report-server-static = [ "x86_64-linux" ];
   };
-in (mergeAttrsMap (dconfig: { ${dconfig } = (withDconfig dconfig).mapTestOn platforms; }) dconfigs)
-   // ((withDconfig null).mapTestOn { stack2nix = supportedSystems; })
+in (mergeAttrsMap (dconfig: { ${dconfig } = (withDconfig dconfig false).mapTestOn platforms; }) dconfigs)
+   // ((withDconfig null false).mapTestOn { stack2nix = supportedSystems; })
    // (rlib.mapTestOn { purescript = supportedSystems; })
+   // (mergeAttrsMap (dconfig: { "${dconfig}-profiled" = (withDconfig dconfig true).mapTestOn platforms; }) dconfigs)
```
Comment (domenkozar) [inline release.nix]: As discussed this needs to be rebased on top of `cardano-sl-1.0-genesis-json2` branch and then we build profiled build only of `cardano-sl-core` on Linux.

**Pull request merged**